tests/components/samsungtv/test_init.py

```diff
@@ -44,34 +41,42 @@
 @pytest.fixture(name="remote")
 def remote_fixture():
     """Patch the samsungctl Remote."""
-    with patch("homeassistant.components.samsungtv.socket") as socket1, patch(
+    with patch(
+        "homeassistant.components.samsungtv.bridge.Remote"
+    ) as remote_class, patch(
         "homeassistant.components.samsungtv.config_flow.socket"
-    ) as socket2, patch("homeassistant.components.samsungtv.config_flow.Remote"), patch(
-        "homeassistant.components.samsungtv.media_player.SamsungRemote"
-    ) as remote:
+    ) as socket1, patch(
+        "homeassistant.components.samsungtv.socket"
+    ) as socket2:
+        remote = mock.Mock()
+        remote.__enter__ = mock.Mock()
+        remote.__exit__ = mock.Mock()
+        remote_class.return_value = remote
         socket1.gethostbyname.return_value = "FAKE_IP_ADDRESS"
         socket2.gethostbyname.return_value = "FAKE_IP_ADDRESS"
         yield remote
 
 
 async def test_setup(hass, remote):
     """Test Samsung TV integration is setup."""
-    await async_setup_component(hass, SAMSUNGTV_DOMAIN, MOCK_CONFIG)
-    await hass.async_block_till_done()
-    state = hass.states.get(ENTITY_ID)
+    with patch("homeassistant.components.samsungtv.bridge.Remote") as remote:
+        await async_setup_component(hass, SAMSUNGTV_DOMAIN, MOCK_CONFIG)
+        await hass.async_block_till_done()
+        state = hass.states.get(ENTITY_ID)
 
-    # test name and turn_on
-    assert state
-    assert state.name == "fake_name"
-    assert (
-        state.attributes[ATTR_SUPPORTED_FEATURES] == SUPPORT_SAMSUNGTV | SUPPORT_TURN_ON
-    )
+        # test name and turn_on
+        assert state
+        assert state.name == "fake_name"
+        assert (
+            state.attributes[ATTR_SUPPORTED_FEATURES]
+            == SUPPORT_SAMSUNGTV | SUPPORT_TURN_ON
+        )
 
-    # test host and port
-    assert await hass.services.async_call(
-        DOMAIN, SERVICE_VOLUME_UP, {ATTR_ENTITY_ID: ENTITY_ID}, True
-    )
-    assert remote.mock_calls[0] == call(REMOTE_CALL)
+        # test host and port
+        assert await hass.services.async_call(
+            DOMAIN, SERVICE_VOLUME_UP, {ATTR_ENTITY_ID: ENTITY_ID}, True
+        )
+        assert remote.call_args == call(REMOTE_CALL)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5152307)